main/inc/lib/api.lib.php

```diff
@@ -9426,6 +9428,28 @@ function api_mail_html(
     }
 
     $mail = new PHPMailer();
+
+    if (isset($platform_email['XOAUTH2_METHOD']) && $platform_email['XOAUTH2_METHOD']) {
```

Comment:
Sounds like it can be simplified to `!empty($platform_email['XOAUTH2_METHOD'])`.